javaslang-benchmark/src/test/java/javaslang/MemoryUsage.java

```diff
@@ -1,37 +1,62 @@
+    private static final DecimalFormat FORMAT = new DecimalFormat("#,##0");
+    private static Map<Integer, LinkedHashSet<Tuple3<CharSeq, CharSeq, CharSeq>>> memoryUsages = TreeMap.empty(); // if forked, this will be reset every time
 
     /** Calculate the occupied memory of different internals */
     static void printAndReset() {
-        for (int size : memoryUsages.keySet()) {
-            System.out.println(String.format("\nfor `%d` elements", size));
-            for (String usages : memoryUsages.get(size).get()) {
-                System.out.println(usages);
+        for (Tuple2<Integer, LinkedHashSet<Tuple3<CharSeq, CharSeq, CharSeq>>> entry : memoryUsages) {
+            final Tuple3<Integer, Integer, Integer> columnSizes = columnSized(entry._1);
+            System.out.println(String.format("\nfor `%d` elements", entry._1));
+            for (Tuple3<CharSeq, CharSeq, CharSeq> stats : entry._2) {
+                final String format = String.format("  %s → %s bytes - %s",
+                        stats._1.padTo(columnSizes._1, ' '),
+                        stats._2.leftPadTo(columnSizes._2, ' '),
+                        stats._3.leftPadTo(columnSizes._3, ' ')
+                );
+                System.out.println(format);
             }
         }
 
         memoryUsages = memoryUsages.take(0); // reset
     }
+    private static Tuple3<Integer, Integer, Integer> columnSized(int size) {
+        return memoryUsages.get(size)
+                .map(ts -> ts.map(t -> t.map(CharSeq::length, CharSeq::length, CharSeq::length))).get()
+                .reduce((t1, t2) -> Tuple.of(max(t1._1, t2._1), max(t1._2, t2._2), max(t1._3, t2._3)));
+    }
 
-    public static void storeMemoryUsages(int elementCount, Object target) {
-        final CharSeq name = CharSeq.of(target.getClass().getPackage().getName()).splitSeq("\\.").map(CharSeq::head).mkCharSeq("", ".", "." + target.getClass().getSimpleName());
-        final String usage = String.format("%s  → %s bytes (%s)",
-                name.padTo(32, ' '),
-                CharSeq.of(DECIMAL_FORMAT.format(sizeOf(target))).leftPadTo(15, ' '),
-                humanReadableUnits(sizeOf(target)));
-        if (!memoryUsages.get(elementCount).getOrElse(List::empty).contains(usage)) {
-            memoryUsages = memoryUsages.put(elementCount, usage);
-        }
+    static void storeMemoryUsages(int elementCount, Object target) {
+        memoryUsages = memoryUsages.put(elementCount, memoryUsages.get(elementCount).getOrElse(LinkedHashSet.empty()).add(Tuple.of(
+                toHumanReadableName(target),
+                FORMAT.format(sizeOf(target)),
+                humanReadableUnits(sizeOf(target))
+        ).map(CharSeq::of, CharSeq::of, CharSeq::of)));
+    }
+    private static HashMap<Pattern, String> names = HashMap.ofEntries(
+            Tuple.of("^java\\.", "Java mutable @ "),
```

Comment:
this is more type-safe, and the tuple aliase will make it more readable in the future :)